File: solutions/day_04/advent4.1.rb
#!/usr/bin/env ruby

def input
  @input ||= STDIN.read.split("\n").map{|row| row.split}
end

def calculate_valid_passphrases
  input.select{|i| i.uniq.count == i.count}.count
end

if __FILE__ == $0
  p calculate_valid_passphrases
end

File: solutions/day_04/advent4.1.ex
#!/usr/bin/env elixir

IO.read(:stdio, :all)
|> String.split("\n")
|> Enum.map(&(&1 |> String.split))
|> Enum.filter(fn(line) ->
  line
  |> Enum.uniq
  |> length == line |> length
end)
|> length
|> IO.puts
